external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaDataConsumer.scala

```diff
@@ -18,222 +18,247 @@
-   *                           poll when `records` is drained.
-   */
-  private case class FetchedData(
-      private var _records: ju.ListIterator[ConsumerRecord[Array[Byte], Array[Byte]]],
-      private var _nextOffsetInFetchedData: Long,
-      private var _offsetAfterPoll: Long) {
-
-    def withNewPoll(
-        records: ju.ListIterator[ConsumerRecord[Array[Byte], Array[Byte]]],
-        offsetAfterPoll: Long): FetchedData = {
-      this._records = records
-      this._nextOffsetInFetchedData = UNKNOWN_OFFSET
-      this._offsetAfterPoll = offsetAfterPoll
-      this
-    }
-
-    /** Whether there are more elements */
-    def hasNext: Boolean = _records.hasNext
-
-    /** Move `records` forward and return the next record. */
-    def next(): ConsumerRecord[Array[Byte], Array[Byte]] = {
-      val record = _records.next()
-      _nextOffsetInFetchedData = record.offset + 1
-      record
-    }
+private[kafka010] case class FetchedData(
+    private var _records: ju.ListIterator[ConsumerRecord[Array[Byte], Array[Byte]]],
+    private var _nextOffsetInFetchedData: Long,
+    private var _offsetAfterPoll: Long) {
+
+  def withNewPoll(
+      records: ju.ListIterator[ConsumerRecord[Array[Byte], Array[Byte]]],
+      offsetAfterPoll: Long): FetchedData = {
+    this._records = records
+    this._nextOffsetInFetchedData = UNKNOWN_OFFSET
+    this._offsetAfterPoll = offsetAfterPoll
+    this
+  }
 
-    /** Move `records` backward and return the previous record. */
-    def previous(): ConsumerRecord[Array[Byte], Array[Byte]] = {
-      assert(_records.hasPrevious, "fetchedData cannot move back")
-      val record = _records.previous()
-      _nextOffsetInFetchedData = record.offset
-      record
-    }
+  /** Whether there are more elements */
+  def hasNext: Boolean = _records.hasNext
 
-    /** Reset the internal pre-fetched data. */
-    def reset(): Unit = {
-      _records = ju.Collections.emptyListIterator()
-    }
+  /** Move `records` forward and return the next record. */
+  def next(): ConsumerRecord[Array[Byte], Array[Byte]] = {
+    val record = _records.next()
+    _nextOffsetInFetchedData = record.offset + 1
+    record
+  }
 
-    /**
-     * Returns the next offset in `records`. We use this to verify if we should check if the
-     * pre-fetched data is still valid.
-     */
-    def nextOffsetInFetchedData: Long = _nextOffsetInFetchedData
+  /** Move `records` backward and return the previous record. */
+  def previous(): ConsumerRecord[Array[Byte], Array[Byte]] = {
+    assert(_records.hasPrevious, "fetchedData cannot move back")
+    val record = _records.previous()
+    _nextOffsetInFetchedData = record.offset
+    record
+  }
 
-    /**
-     * Returns the next offset to poll after draining the pre-fetched records.
-     */
-    def offsetAfterPoll: Long = _offsetAfterPoll
+  /** Reset the internal pre-fetched data. */
+  def reset(): Unit = {
+    _records = ju.Collections.emptyListIterator()
```

Comment:
Should reflect #22507 when #22507 is merged.

Reply:
Is it solved?

Reply:
I'll check it again. I should have used "Resolve conversation".

Reply:
I think it's solved.